sentinelhub/config.py

```diff
@@ -11,7 +11,116 @@
 ConfigDict = Dict[str, Union[str, int, float]]
 
 
-class SHConfig:
+class _BaseSHConfig:
+    """A base class for the sentinelhub-py package configuration class.
+
+    Only contains defaults and custom getter methods.
+    """
+
+    CREDENTIALS = {
+        "instance_id",
+        "sh_client_id",
+        "sh_client_secret",
+        "aws_access_key_id",
+        "aws_secret_access_key",
+        "aws_session_token",
+    }
+    OTHER_PARAMS = {
+        "sh_base_url",
+        "sh_auth_base_url",
+        "geopedia_wms_url",
+        "geopedia_rest_url",
+        "aws_metadata_url",
+        "aws_s3_l1c_bucket",
+        "aws_s3_l2a_bucket",
+        "opensearch_url",
+        "max_wfs_records_per_query",
+        "max_opensearch_records_per_query",
+        "max_download_attempts",
+        "download_sleep_time",
+        "download_timeout_seconds",
+        "number_of_download_processes",
+    }
+    CONFIG_PARAMS = CREDENTIALS.union(OTHER_PARAMS)
+
+    def __init__(self) -> None:
+        self.instance_id: str = ""
+        self.sh_client_id: str = ""
+        self.sh_client_secret: str = ""
+        self.sh_base_url: str = "https://services.sentinel-hub.com"
+        self.sh_auth_base_url: str = "https://services.sentinel-hub.com"
+        self.geopedia_wms_url: str = "https://service.geopedia.world"
+        self.geopedia_rest_url: str = "https://www.geopedia.world/rest"
+        self.aws_access_key_id: str = ""
+        self.aws_secret_access_key: str = ""
+        self.aws_session_token: str = ""
+        self.aws_metadata_url: str = "https://roda.sentinel-hub.com"
+        self.aws_s3_l1c_bucket: str = "sentinel-s2-l1c"
+        self.aws_s3_l2a_bucket: str = "sentinel-s2-l2a"
+        self.opensearch_url: str = "http://opensearch.sentinel-hub.com/resto/api/collections/Sentinel2"
+        self.max_wfs_records_per_query: int = 100
+        self.max_opensearch_records_per_query: int = 500  # pylint: disable=invalid-name
+        self.max_download_attempts: int = 4
+        self.download_sleep_time: float = 5.0
+        self.download_timeout_seconds: float = 120.0
+        self.number_of_download_processes: int = 1
```

Comment:
What that cause value changes to be reflected across all instances of the class?

Reply:
No, if you use them with `self.instance_id` this will only modify the value that is stored on class instance. Only if you would be changing `SHConfig.instance_id` then that would change it globally. But we don't do that.

Reply:
but the users technically could, right?

Reply:
Yeah, technically they could. Ok, let's leave it as it is so that they won't.

Reply:
sometimes i am terrified of users